src/ModbusClientTCPasync.cpp

```diff
@@ -111,12 +107,11 @@ Error ModbusClientTCPasync::addRequest(ModbusMessage msg, uint32_t token) {
 bool ModbusClientTCPasync::addToQueue(int32_t token, ModbusMessage request) {
   // Did we get one?
   if (request) {
-    #if USE_MUTEX
-    lock_guard<mutex> qLockGuard(qLock);
-    #endif
+    LOCK_GUARD(lock1, qLock);
     if (txQueue.size() + rxQueue.size() < MTA_qLimit) {
       HEXDUMP_V("Enqueue", request.data(), request.size());
       RequestEntry *re = new RequestEntry(token, request);
+      re->sentTime = millis(); //@Miq1 Why do I need to put this here?
```

Comment:
Don't know why I need to add in here, the statement below doesn't work

Reply:
You had that there already, just twice. Please refer to the master branch, it should still be visible there.